Serve test-repo RPMs over a super simple http server

sidcha opened the pull request:
Now, test-repo is served at http://localhost:8080/ on the test/dev
machine. Hopefully, this get tests closer to actual usage scenario.

Also, fix tests/test_downgrade.py: test_downgrade_no_arg() and add
new key "distribution" to utils.config so tests can perform some
distribution specifc tests (or outcome asserts).

Signed-off-by: Siddharth Chandrasekaran <csiddharth@vmware.com>

sidcha committed:
Serve test-repo RPMs over a super simple http server

Now, test-repo is served at http://localhost:8080/ on the test/dev
machine. Hopefully, this get tests closer to actual usage scenario.

Also, fix tests/test_downgrade.py: test_downgrade_no_arg() and add
new key "distribution" to utils.config so tests can perform some
distribution specifc tests (or outcome asserts).

Signed-off-by: Siddharth Chandrasekaran <csiddharth@vmware.com>

tapakund commented on pytests/repo/setup-repo.sh:
```diff
@@ -88,7 +88,7 @@ cat << EOF > ${PUBLISH_PATH}/metalink
    </verification>
    <resources maxconnections="1">
-    <url protocol="file" type="file" location="IN" preference="100">file://${PUBLISH_PATH}/repodata/repomd.xml</url>
+    <url protocol="file" type="file" location="IN" preference="100">http://localhost:8080/photon-test/repodata/repomd.xml</url>
```
Sid, pls change the protocol and type to http.

sidcha replied:
Done. Thanks!

tapakund replied:
I don't see the changes reflected in the commit.

sshedi commented on pytests/tests/test_downgrade.py:
```diff
@@ -22,15 +22,8 @@ def teardown_test(utils):
-    except Exception as e:
-        # Default to Photon
-        assert(ret['retval'] == 8)
+    ret = utils.run(['tdnf', 'downgrade'])
```
The reason I used different config for Photon was, we install glibc-debuginfo in test environment and with that 'tdnf downgrade' should try to downgrade everything and with assume assumeno we should get 1602 as retval.
TBH, I didn't knew what 'tdnf downgrade' without args will do and @keerthanakalyan gave me insight on how it is supposed to work so I tried to achieve it.

sidcha replied:
Ohh, I see. Then we must probably call it `test_downgrade_everything` and create some new dummy spec files that simulate this situation. For now, let the `test_downgrade_no_args` be consistent with it's naming (with other such tests everywhere). We'll put in a TODO to add the other test.

sidcha committed:
Disable valgrind checks in fedora CI builds

Looks like the docker image for fedora, pulled by travis, has some
issue with valgrind. Local checks on fedora image does succeed; so
I have no idea as to what is causing this issue. For now, skipping
memchecks when $DIST=fedora. This is still okay as memchecks run on
photon.

This commit also cleans up TestUtils::check_valgrind() as levels of
indents were getting unmanageable.

Signed-off-by: Siddharth Chandrasekaran <csiddharth@vmware.com>
